### DISCUSSION TITLE
Specyfying an image format

### DISCUSSION THREAD

**Discussion by TzOk83** [keirf/greaseweazle#103] [Q&A]
I can't find how to explicitly specify an image format.

```
gw read --format IPF test.ipf
** FATAL ERROR:
Unknown format 'IPF'
```

```
gw read --format ADF test.adf
** FATAL ERROR:
Unknown format 'ADF'
```

```
gw read --format HFE test.hfe
** FATAL ERROR:
Unknown format 'HFE'
```

How to get a list of supported image formats/--format option parameters?

**Comment by ourIThome** [comment]
Have you tried:

gw read Folder name/track.raw
gw read test.adf
gw read test.hfe

To save as ipf ... Using the raw files, you will need to use Keir disk utilities.

**Comment by barbeque** [comment]
Another option is to read as HFE and then convert it using the free HxC HxCFloppyEmulator utility (for Mac and Windows; there may be a Linux version but I haven't been able to find it yet.) https://hxc2001.com/download/floppy_drive_emulator/

**Reply by M1kerochip** [reply]
The disk format is determined by the file extension.

GW can only write .ipf formats, it can't create them.

To see the extensions, look in util.py

You don't need to read it as a HFE to convert to other formats using HxC. HxC also can't convert to .ipf
